src/Swarm/App.hs

```diff
@@ -105,3 +119,30 @@ enablePasteMode = do
   when (V.supportsMode output V.BracketedPaste) $
     liftIO $
       V.setMode output V.BracketedPaste True
+
+version :: String
```

Comment:
Perhaps this could be `Maybe String`, where a local dirty git status should have a Nothing version. That way we can skip the upstreamVersion query when developing swarm.

Reply:
Good point about skipping the check during development, I will add a check for that.

Why would the status be dirty though?

We have tags both locally and when building in CI, so you will not tell the difference by it being empty.

Instead I think I will have to use the hash that is appended to the tag name and compare it to current commit.

Reply:
@TristanCacqueray well, it did not go as I hoped, but I made another function which gives `Maybe` and another that gives `Either`. Hopefully, that will make using the results easier. :slightly_smiling_face:

Reply:
I see, that's annoying... How about checking the branch name, if it is `main`, do the new version notification, otherwise assume it is a develop copy?

It might be good to check the values for a `cabal install` build, if I remember correctly, the .git content is not available in that situation.

Reply:
> @TristanCacqueray ...if it's `main`...

That is indeed the approach I have taken. :laughing: 

> It might be good to check the values for a `cabal install` build, if I remember correctly, the .git content is not available in that situation.

The `cabal install` binary is indeed problematic:
```
swarm version
Swarm game - pre-alpha version
"GHEGitRunFailed \".\" [\"rev-parse\",\"--show-toplevel\"] (ExitFailure 128) \"\" \"fatal: not a git repository (or any parent up to mount point /)\\nStopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).\\n\""
```
Maybe we could include the `.git` folder for the build only in cabal? :thinking:

Reply:
Or I could use the readily available `Paths_swarm.version` and save myself the trouble! :grinning: